include/zephyr/net/dhcpv4.h

```diff
@@ -64,7 +64,8 @@ enum net_dhcpv4_msg_type {
 	NET_DHCPV4_MSG_TYPE_INFORM	= 8,
 };
 
-#ifdef CONFIG_NET_DHCPV4_OPTION_CALLBACKS
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS) ||                                                 \
```

Comment:
I just wonder if it'd make sense to make `CONFIG_NET_DHCPV4_OPTION_CALLBACKS_VENDOR_SPECIFIC` dependent on `CONFIG_NET_DHCPV4_OPTION_CALLBACKS` (so that you need to enable option callbacks to get vendor-specific extensions). That should reduce ifdef slicing of the code a bit, that long `if defined(this) || defined(that)` condition looks a bit worrying.

Reply:
Well, on the one hand, the functionality does not really depend on option callbacks being enabled. On the other hand I agree that from a user standpoint encapsulated option callbacks are just a specific type of option callbacks, even if their parsing is done separately.
Overall, I am also in favor of making encapsulated vendor options a suboption in Kconfig.